### PR TITLE
Add TA-65 PCB to QMK

### PR DESCRIPTION
**Pull request by yanfali**
Adds new PCB from @Maartenwut to QMK main repo, with 4 different user friendly layouts (ansi, ansi_tsangan, iso and iso_tsangan). 

## Description
Submitting on behalf of Maarten. Main code is from Maarten, layouts were added by me. ISO layout was tested by ISO_Returns.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ta65/config.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
```
Comment (fauxpark) [inline keyboards/ta65/config.h]: This block can be removed or commented out, as it's the default key combination already defined in QMK.

File: keyboards/ta65/rules.mk [hunk 2]
```diff
@@ -0,0 +1,65 @@
+
```
Comment (fauxpark) [inline keyboards/ta65/rules.mk]:
```suggestion
```

File: keyboards/ta65/rules.mk [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+#   Teensy halfKay   512
+#   Teensy++ halfKay 1024
+#   Atmel DFU loader 4096
+#   LUFA bootloader  4096
+#   USBaspLoader     2048
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (fauxpark) [inline keyboards/ta65/rules.mk]:
```suggestion
```

`BOOTLOADER` should automatically set this.

File: keyboards/ta65/readme.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+Keyboard Maintainer: QMK Community<br>
+Hardware Supported: TA-65 PCB<br>
+Hardware Availability: [maartenwut.com](https://maartenwut.com/product/ta-65/)<br>
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ta65:default
+
+See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (fauxpark) [inline keyboards/ta65/readme.md]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/ta65/keymaps/maartenwut/config.h [hunk 5]
```diff
@@ -0,0 +1 @@
+#define RETRO_TAPPING
```
Comment (drashna) [inline keyboards/ta65/keymaps/maartenwut/config.h]: This should have an include guard for this. 

```suggestion
#pragma once 

#define RETRO_TAPPING
```

Comment (yanfali) [reply keyboards/ta65/keymaps/maartenwut/config.h]: thanks Drashna. I wasn't sure if it needed it because of how it's included

**Comment by yanfali** [comment]
@drashna good to

**Comment by yanfali** [comment]
> I think we will soon need a martenwuut parent directory

Wut you talking about Merlin?

**Comment by Maartenwut** [comment]
> I think we will soon need a martenwuut parent directory

Might be a good idea.

**Pull request merged**